### DISCUSSION TITLE
Using Kratos without specifying a hostname

### DISCUSSION THREAD

**Discussion by velovix** [ory/kratos#1618] [Q&A] [answered]
I'm currently working on integrating Kratos with a product that is most often deployed in local networks. Our users are accustomed to being able to access the server from a variety of hostnames, including localhost, a local IP address, and occasionally a public IPv4 address. As far as I can tell, Kratos requires me to explicitly specify in the configuration file the hostname that the server will be accessed on, which doesn't fit with this workflow. Is there a way to avoid this? Ideally, I would want Kratos to just use whatever hostname the client is currently accessing it on.

One idea I briefly considered was putting an HTTP proxy like Nginx in front of Kratos and doing some trickery to change the apparent hostname of the request. Has anyone experimented with a solution like that?

**Comment by aeneasr** [comment, marked as answer]
The Nginx route is probably the right one. Ory Kratos does not really support multi-domain environments well yet.

**Reply by velovix** [reply]
That's unfortunate, but thank you for the answer. I'll provide an update here with my solution if I end up going this route.

**Reply by aeneasr** [reply]
Great, I think people would appreciate that! You can also look at ory/cli -> proxy. We're doing something similar there
